src/plugin/file/kml/kmlparser.js

```diff
@@ -1590,17 +1654,22 @@ plugin.file.kml.KMLParser.prototype.mapStyleToConfig_ = function(style) {
  * Finds the first instance of a style id on the style stack
  * @param {string} id The style id
  * @param {boolean=} opt_highlight Whether to check the highlight style map
+ * @param {boolean=} opt_otherStyle Whether to check externally parsed styles.
+
  * @return {Array<Object>} The style configs, or null if not found
  * @private
  */
-plugin.file.kml.KMLParser.prototype.findStyle_ = function(id, opt_highlight) {
+plugin.file.kml.KMLParser.prototype.findStyle_ = function(id, opt_highlight, opt_otherStyle) {
```

Comment:
This function is a little awkward, given the parameter train to adjust which map is used. Instead, we should extract the bit that determines the id into a function and use the result to retrieve values from the appropriate map directly.